inference-engine/tests/functional/plugin/cpu/single_layer_tests/tensor_iterator.cpp

```diff
@@ -0,0 +1,118 @@
+    }
+
+protected:
+    void SetUp() override {
+        std::vector<InputShape> shapes;
+        ngraph::op::RecurrentSequenceDirection direction;
+        ElementType inType;
+        std::tie(shapes, direction, inType) = this->GetParam();
+
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        init_input_shapes({shapes});
+
+        const size_t sequence_axis = 1;
+        auto tensor_iterator = std::make_shared<ngraph::opset5::TensorIterator>();
+        auto params = ngraph::builder::makeDynamicParams(inType, inputDynamicShapes);
+
+        ngraph::ParameterVector body_params;
+        for (size_t i = 0; i < shapes.size(); i++) {
+            ngraph::PartialShape shape = shapes[i].first;
+            shape[sequence_axis] = 1;
+            auto paramNode = std::make_shared<ngraph::opset1::Parameter>(inType, shape);
+            body_params.push_back(paramNode);
+        }
+        auto hswish = ngraph::builder::makeActivation(body_params[0], inType, ngraph::helpers::HSwish);
+        auto relu = ngraph::builder::makeActivation(body_params[1], inType, ngraph::helpers::Relu);
+        auto add = std::make_shared<ngraph::opset1::Add>(hswish, relu);
+
+        auto body = std::make_shared<ngraph::Function>(ngraph::OutputVector{add}, body_params, "body");
+        tensor_iterator->set_function(body);
+
+        if (direction == ngraph::op::RecurrentSequenceDirection::FORWARD) {
+            tensor_iterator->set_sliced_input(body_params[0], params[0], 0, 1, 1, -1, sequence_axis);
+            tensor_iterator->set_sliced_input(body_params[1], params[1], 0, 1, 1, -1, sequence_axis);
+            tensor_iterator->get_concatenated_slices(add, 0, 1, 1, -1, sequence_axis);
+        } else if (direction == ngraph::op::RecurrentSequenceDirection::REVERSE) {
+            tensor_iterator->set_sliced_input(body_params[0], params[0], -1, -1, 1, 0, sequence_axis);
+            tensor_iterator->set_sliced_input(body_params[1], params[1], -1, -1, 1, 0, sequence_axis);
+            tensor_iterator->get_concatenated_slices(add, -1, -1, 1, 0, sequence_axis);
+        } else {
+            NGRAPH_CHECK(false, "Bidirectional case is not supported.");
+        }
+
+        function = std::make_shared<ngraph::Function>(ngraph::OutputVector{tensor_iterator->output(0)}, params);
+    }
+};
+
+TEST_P(TensorIteratorCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+
+const std::vector<ElementType> inputPrecisions = {
+        ElementType::f32,
+        ElementType::bf16
+};
+
+std::vector<ngraph::op::RecurrentSequenceDirection> direction = {ngraph::op::RecurrentSequenceDirection::FORWARD,
+                                                                 ngraph::op::RecurrentSequenceDirection::REVERSE};
+std::vector<InputShape> inputs = {
+        {{-1, 12, -1}, {{10, 12, 10}, {10, 12, 10}, {1, 12, 2}, {5, 12, 3}}},
+        {{-1, 12, -1}, {{1, 12, 1}, {1, 12, 1}, {5, 12, 2}, {5, 12, 3}}},
```

Comment:
Could you please add test cases with defined ranges?

Reply:
Of course, I could. Done, Thanks!